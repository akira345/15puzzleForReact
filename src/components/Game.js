import React, { useState } from 'react';
import ShowStatus from './ShowStatus';
import ShowLevel from './ShowLevel';
import Board from './Board';
import GameController from './GameControler';
import Puzzle from '../util/puzzle';

// メインのゲームコンポーネント
const Game = () => {

  // 盤面数設定
  const [ puzzle, setPuzzle ] = useState( new Puzzle( 9 ) );
  console.log( "初期値" );
  console.log( puzzle.emptyCellNo );
  console.log( puzzle.totalCount );
  // 初期データセット
  const answerDatas = puzzle.answerDatas;
  // ゲーム初期データセット
  const gameDatas = puzzle.gameDatas;
  const sideSize = puzzle.sideSize;

  const handleSubmit = ( event ) => {
    // 一旦デフォルトのイベントを止める（いるのか分からん）
    event.preventDefault();
    if ( event.target.submit.value === 'スタート' ) {
      // 設定された難易度をセット
      puzzle.difficult = event.target.difficult.value;
      // ランダム作成
      puzzle.generateGameDatas();
      console.log( puzzle.gameDatas );
      // セット
      // useStateはイミュータブルじゃないとダメ
      // とりあえず今はインスタンスをコピーしてお茶を濁す
      const clone = Object.assign( Object.create( Object.getPrototypeOf( puzzle ) ), puzzle );
      setPuzzle( clone );
    } else {
      // ゲーム初期化
      setPuzzle( new Puzzle( 9 ) );
    }

  };

  return (
    <>
      <ShowLevel
        difficult={ puzzle.difficult }
      />        <ShowStatus
        isComplate={ puzzle.isComplete() }
      />
      <table border="0" >
        <tbody>
          <tr>
            <td>
              <Board
                datas={ gameDatas }
                sideSize={ sideSize }
              />
            </td>
            <td>　</td>
            <td>
              <Board
                datas={ answerDatas }
                sideSize={ sideSize }
              />
            </td>
          </tr>
        </tbody>
      </table>
      <GameController
        handleSubmit={ handleSubmit }
        isComplate={ puzzle.isComplete() }
      />
    </>
  );
};

export default Game;