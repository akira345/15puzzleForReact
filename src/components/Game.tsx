import React, { useState, FC, useRef } from "react";
import ShowStatus from "./ShowStatus";
import ShowLevel from "./ShowLevel";
import Board from "./Board";
import GameController from "./GameControler";
import Puzzle from "../util/puzzle";

/**
 * 強制的に再描画する関数を返すカスタムhook
 */
const useForceRender = () => {
  // つかわないのでアンスコにしている
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const [_, set] = useState({});
  return () => set({});
}

// メインのゲームコンポーネント
const Game: FC = () => {
  const forceRender = useForceRender();
  // 盤面数設定
  // 難易度の選択状態
  const [difficult, setDifficult] = useState(puzzle.difficult);
  const puzzleRef = useRef<Puzzle>(new Puzzle(9));
  const puzzle = puzzleRef.current;
  // 初期データセット
  const answerDatas = puzzle.answerDatas;
  const gameDatas = puzzle.gameDatas;
  const sideSize = puzzle.sideSize;

  const handleDifficultChange = (difficult: string) => {
    setDifficult(difficult);
  }

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    // 一旦デフォルトのイベントを止める（いるのか分からん）
    event.preventDefault();

    if (puzzle.isComplete()) {
      // 設定された難易度をセット
      puzzle.difficult = difficult;
      // ゲームデータ作成
      puzzle.generateGameDatas();
      // 動かせるセルをセット
      puzzle.setMove();
    } else {
      // ギブアップ
      // ゲーム初期化
      puzzleRef.current = new Puzzle(9);
    }
    forceRender();
  };
  // 空白セルクリックハンドラ
  const handleLink = (
    event: React.MouseEvent<HTMLImageElement, MouseEvent>,
    moveIndex: number
  ) => {
    // 一旦デフォルトのイベントを止める（いるのか分からん）
    event.preventDefault();
    // 空白セルを動かす
    puzzle.move(moveIndex);
    forceRender();
    // ゲームがクリアされていなければ、次の動かせるセルをセットする
    if (!puzzle.isComplete()) {
      // 動かせるセルをセット
      puzzle.setMove();
    }
  };

  return (
    <>
      <ShowLevel difficult={puzzle.difficult} />{" "}
      <ShowStatus isComplate={puzzle.isComplete()} />
      <table
        style={{
          border: "none",
        }}
      >
        <tbody>
          <tr>
            <td>
              <Board
                datas={gameDatas}
                sideSize={sideSize}
                moves={puzzle.moves}
                handleLink={handleLink}
              />
            </td>
            <td>　</td>
            <td>
              <Board
                datas={answerDatas}
                sideSize={sideSize}
                moves={[]}
                handleLink={() => {}}
              />
            </td>
          </tr>
        </tbody>
      </table>
      <GameController
        difficult={difficult}
        handleSubmit={handleSubmit}
        isComplate={puzzle.isComplete()}
        onDifficultChange={handleDifficultChange}
      />
    </>
  );
};

export default Game;
