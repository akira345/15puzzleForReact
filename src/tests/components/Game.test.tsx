import React from "react";
import { render, fireEvent, getByText } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Game from "../../components/Game";
const { mockRandom, resetMockRandom } = require("jest-mock-random");

describe("15パズルテスト", () => {
  it("初期画面テスト", () => {
    const { asFragment } = render(<Game />);
    // スナップショットを取る
    expect(asFragment()).toMatchSnapshot();
  });
  it("ゲームスタート", () => {
    // 乱数のモック作成
    mockRandom([0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]);

    const { asFragment, getByLabelText, getByTestId } = render(<Game />);
    // 難易度Easyを選択
    const radio = getByLabelText("やさしい") as HTMLInputElement;
    fireEvent.click(radio);
    // 値が変わったか？
    expect(radio).toBeChecked();
    expect(radio.value).toBe("Easy");

    // IDを付けたフォームのサブミットを呼び出す
    fireEvent.submit(getByTestId("control-form"));
    // スナップショットを取る
    expect(asFragment()).toMatchSnapshot();
    // モックを戻す
    resetMockRandom();
  });
    it("ゲームスタート", () => {
    // 乱数のモック作成
    mockRandom([0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]);

    const { asFragment, getByLabelText, getByTestId } = render(<Game />);
    // 難易度Easyを選択
    const radio = getByLabelText("やさしい") as HTMLInputElement;
    fireEvent.click(radio);
    // 値が変わったか？
    expect(radio).toBeChecked();
    expect(radio.value).toBe("Easy");

    // IDを付けたフォームのサブミットを呼び出す
    fireEvent.submit(getByTestId("control-form"));
    // スナップショットを取る
    expect(asFragment()).toMatchSnapshot();
    // モックを戻す
    resetMockRandom();
  });
});
