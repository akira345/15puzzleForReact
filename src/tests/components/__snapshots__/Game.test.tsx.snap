// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`15パズルテスト ゲームスタート 1`] = `
<DocumentFragment>
  やさしい 未完成
  <table>
    <tbody>
      <tr>
        <td>
          <table
            style="border-width: 1px; border-color: #000000; border-style: solid;"
          >
            <tbody>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="1"
                    src="/gif/1.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="2"
                    src="/gif/2.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="3"
                    src="/gif/3.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="4"
                    src="/gif/4.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="5"
                    src="/gif/5.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="9"
                    src="/gif/9.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="7"
                    src="/gif/7.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="8"
                    src="/gif/8.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="6"
                    src="/gif/6.gif"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </td>
        <td>
          　
        </td>
        <td>
          <table
            style="border-width: 1px; border-color: #000000; border-style: solid;"
          >
            <tbody>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="1"
                    src="/gif/1.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="2"
                    src="/gif/2.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="3"
                    src="/gif/3.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="4"
                    src="/gif/4.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="5"
                    src="/gif/5.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="6"
                    src="/gif/6.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="7"
                    src="/gif/7.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="8"
                    src="/gif/8.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="9"
                    src="/gif/9.gif"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
  <form
    data-testid="control-form"
  >
    <input
      data-testid="control-submit"
      name="submit"
      type="submit"
      value="ギブアップ"
    />
  </form>
</DocumentFragment>
`;

exports[`15パズルテスト ゲームスタート後ギブアップ 1`] = `
<DocumentFragment>
  普通 完成
  <table>
    <tbody>
      <tr>
        <td>
          <table
            style="border-width: 1px; border-color: #000000; border-style: solid;"
          >
            <tbody>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="1"
                    src="/gif/1.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="2"
                    src="/gif/2.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="3"
                    src="/gif/3.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="4"
                    src="/gif/4.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="5"
                    src="/gif/5.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="6"
                    src="/gif/6.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="7"
                    src="/gif/7.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="8"
                    src="/gif/8.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="9"
                    src="/gif/9.gif"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </td>
        <td>
          　
        </td>
        <td>
          <table
            style="border-width: 1px; border-color: #000000; border-style: solid;"
          >
            <tbody>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="1"
                    src="/gif/1.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="2"
                    src="/gif/2.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="3"
                    src="/gif/3.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="4"
                    src="/gif/4.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="5"
                    src="/gif/5.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="6"
                    src="/gif/6.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="7"
                    src="/gif/7.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="8"
                    src="/gif/8.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="9"
                    src="/gif/9.gif"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
  <form
    data-testid="control-form"
  >
    <label>
      <input
        name="difficult"
        type="radio"
        value="Easy"
      />
      やさしい
    </label>
    <label>
      <input
        checked=""
        name="difficult"
        type="radio"
        value="Normal"
      />
      普通
    </label>
    <label>
      <input
        name="difficult"
        type="radio"
        value="Hard"
      />
      難しい
    </label>
    <br />
    <input
      data-testid="control-submit"
      name="submit"
      type="submit"
      value="スタート"
    />
  </form>
</DocumentFragment>
`;

exports[`15パズルテスト セルを動かす(完成) 1`] = `
<DocumentFragment>
  やさしい 完成
  <table>
    <tbody>
      <tr>
        <td>
          <table
            style="border-width: 1px; border-color: #000000; border-style: solid;"
          >
            <tbody>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="1"
                    src="/gif/1.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="2"
                    src="/gif/2.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="3"
                    src="/gif/3.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="4"
                    src="/gif/4.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="5"
                    src="/gif/5.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="6"
                    src="/gif/6.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="7"
                    src="/gif/7.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="8"
                    src="/gif/8.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="9"
                    src="/gif/9.gif"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </td>
        <td>
          　
        </td>
        <td>
          <table
            style="border-width: 1px; border-color: #000000; border-style: solid;"
          >
            <tbody>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="1"
                    src="/gif/1.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="2"
                    src="/gif/2.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="3"
                    src="/gif/3.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="4"
                    src="/gif/4.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="5"
                    src="/gif/5.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="6"
                    src="/gif/6.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="7"
                    src="/gif/7.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="8"
                    src="/gif/8.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="9"
                    src="/gif/9.gif"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
  <form
    data-testid="control-form"
  >
    <label>
      <input
        checked=""
        name="difficult"
        type="radio"
        value="Easy"
      />
      やさしい
    </label>
    <label>
      <input
        name="difficult"
        type="radio"
        value="Normal"
      />
      普通
    </label>
    <label>
      <input
        name="difficult"
        type="radio"
        value="Hard"
      />
      難しい
    </label>
    <br />
    <input
      data-testid="control-submit"
      name="submit"
      type="submit"
      value="スタート"
    />
  </form>
</DocumentFragment>
`;

exports[`15パズルテスト セルを動かす(未完成) 1`] = `
<DocumentFragment>
  やさしい 未完成
  <table>
    <tbody>
      <tr>
        <td>
          <table
            style="border-width: 1px; border-color: #000000; border-style: solid;"
          >
            <tbody>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="1"
                    src="/gif/1.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="2"
                    src="/gif/2.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="9"
                    src="/gif/9.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="4"
                    src="/gif/4.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="5"
                    src="/gif/5.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="3"
                    src="/gif/3.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="7"
                    src="/gif/7.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="8"
                    src="/gif/8.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="6"
                    src="/gif/6.gif"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </td>
        <td>
          　
        </td>
        <td>
          <table
            style="border-width: 1px; border-color: #000000; border-style: solid;"
          >
            <tbody>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="1"
                    src="/gif/1.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="2"
                    src="/gif/2.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="3"
                    src="/gif/3.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="4"
                    src="/gif/4.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="5"
                    src="/gif/5.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="6"
                    src="/gif/6.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="7"
                    src="/gif/7.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="8"
                    src="/gif/8.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="9"
                    src="/gif/9.gif"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
  <form
    data-testid="control-form"
  >
    <input
      data-testid="control-submit"
      name="submit"
      type="submit"
      value="ギブアップ"
    />
  </form>
</DocumentFragment>
`;

exports[`15パズルテスト 初期画面テスト 1`] = `
<DocumentFragment>
  普通 完成
  <table>
    <tbody>
      <tr>
        <td>
          <table
            style="border-width: 1px; border-color: #000000; border-style: solid;"
          >
            <tbody>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="1"
                    src="/gif/1.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="2"
                    src="/gif/2.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="3"
                    src="/gif/3.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="4"
                    src="/gif/4.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="5"
                    src="/gif/5.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="6"
                    src="/gif/6.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="7"
                    src="/gif/7.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="8"
                    src="/gif/8.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="9"
                    src="/gif/9.gif"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </td>
        <td>
          　
        </td>
        <td>
          <table
            style="border-width: 1px; border-color: #000000; border-style: solid;"
          >
            <tbody>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="1"
                    src="/gif/1.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="2"
                    src="/gif/2.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="3"
                    src="/gif/3.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="4"
                    src="/gif/4.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="5"
                    src="/gif/5.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="6"
                    src="/gif/6.gif"
                  />
                </td>
              </tr>
              <tr>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="7"
                    src="/gif/7.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="8"
                    src="/gif/8.gif"
                  />
                </td>
                <td
                  style="border-width: 1px; border-color: #000000; border-style: solid;"
                >
                  <img
                    alt="9"
                    src="/gif/9.gif"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
  <form
    data-testid="control-form"
  >
    <label>
      <input
        name="difficult"
        type="radio"
        value="Easy"
      />
      やさしい
    </label>
    <label>
      <input
        checked=""
        name="difficult"
        type="radio"
        value="Normal"
      />
      普通
    </label>
    <label>
      <input
        name="difficult"
        type="radio"
        value="Hard"
      />
      難しい
    </label>
    <br />
    <input
      data-testid="control-submit"
      name="submit"
      type="submit"
      value="スタート"
    />
  </form>
</DocumentFragment>
`;
